### PR TITLE
Tablet ui -- misc changes + landscape mode

### PR DESCRIPTION
**Pull request by sethalves**
- remove code that allowed tablet to be visible to others
- add tablet.setLandscape() javascript call
- attempt to fix a hang-on-shutdown

### CODE REVIEW

**Comment by sethalves** [comment]
to test:
- generally exercise the tablet
- enable tablet in 2d mode
- open console
- run: ```javascript
Tablet.getTablet('com.highfidelity.interface.tablet.system').landscape = true;
```
- observe that the tablet is now wider than tall
- run: ```javascript
Tablet.getTablet('com.highfidelity.interface.tablet.system').landscape = false;
```
- observe that the tablet is now taller than wide

**Comment by hifi-gustavo** [comment]
Build console output for commit: c62b07d26f5bea4c246313f3ae4abfcef40e8382 available at: https://gist.github.com/4f85bc103cdaf1cee2fed7452a2a40c4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10075-c62b07d.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10075-c62b07d.dmg

File: scripts/system/libraries/WebTablet.js [hunk 1]
```diff
@@ -105,20 +105,23 @@ WebTablet = function (url, width, dpi, hand, clientOnly, location) {
...
         this.dpi = DEFAULT_DPI * (DEFAULT_WIDTH / this.width);
     }
 
-    var modelURL;
-    if (Settings.getValue("tabletVisibleToOthers")) {
-        modelURL = TABLET_MODEL_PATH;
-    } else {
-        modelURL = LOCAL_TABLET_MODEL_PATH;
-    }
+    this.getDimensions = function() {
```
Comment (hyperlogic) [inline scripts/system/libraries/WebTablet.js]: So hey, I'd rather this method be on the WebTablet constructor prototype:

    WebTablet.prototype.getDimensions = function () {
        //
    }

But this is just a personal preference

File: scripts/system/libraries/WebTablet.js [hunk 2]
```diff
@@ -128,10 +131,18 @@ WebTablet = function (url, width, dpi, hand, clientOnly, location) {
         userData: JSON.stringify({
             "grabbableKey": {"grabbable": true}
         }),
-        dimensions: {x: this.width, y: this.height, z: this.depth},
+        dimensions: this.getDimensions(),
         parentID: AVATAR_SELF_ID
     };
 
+    this.getTabletTextureResolution = function() {
```
Comment (hyperlogic) [inline scripts/system/libraries/WebTablet.js]: Similarly, this should be attached to the WebTablet.prototype.

File: scripts/system/libraries/WebTablet.js [hunk 3]
```diff
@@ -203,23 +209,30 @@ WebTablet = function (url, width, dpi, hand, clientOnly, location) {
         }
     };
 
+    this.setLandscape = function(newLandscapeValue) {
```
Comment (hyperlogic) [inline scripts/system/libraries/WebTablet.js]: Same here

File: scripts/system/libraries/WebTablet.js [hunk 4]
```diff
@@ -203,23 +209,30 @@ WebTablet = function (url, width, dpi, hand, clientOnly, location) {
...
+    };
+
     this.state = "idle";
 
     this.getRoot = function() {
         return Entities.getWebViewRoot(_this.tabletEntityID);
     };
 
     this.getLocation = function() {
-        if (this.tabletIsOverlay) {
```
Comment (hyperlogic) [inline scripts/system/libraries/WebTablet.js]: same here

**Comment by hifi-gustavo** [comment]
Build console output for commit: 715493922e2f9990d2e95ad996af8ff5f0b3e648 available at: https://gist.github.com/4f85bc103cdaf1cee2fed7452a2a40c4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10075-7154939.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10075-7154939.dmg

**Comment by claudioshg** [comment]
testing

**Comment by claudioshg** [comment]
### Desktop PC - Windows 10 Home - CPU 4.0GHz Intel i7 - GTX1070 - RAM 16GB - Tested with Oculus Rift

Tested: Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10075-7154939.exe

- generally exercise the tablet: PASS
- run: Tablet.getTablet('com.highfidelity.interface.tablet.system').landscape = true; PASS
- run: Tablet.getTablet('com.highfidelity.interface.tablet.system').landscape = false; FAIL

Fails:
- When I run the line, the tablet stays the same. It doesn't stretch the tablet like the first line.
pr10075

**Comment by sethalves** [comment]
@claudioshg -- maybe I'm not understanding.  the first line (ending with true) should stretch the tablet.  the second line (ending with false) should "unstretch" it, and return it to its original shape.  is this not what happened?

**Comment by Nex-Pro** [comment]
@seth I'm just talking @claudioshg thru True and False

**Comment by sethalves** [comment]
oh -- I see.  the test plan wasn't clear.  he was expecting it to get "taller than wide" by growing vertically.

**Comment by Nex-Pro** [comment]
@sethalves true

**Comment by hifi-gustavo** [comment]
Build console output for commit: 43e1cec8a3e06a2281ce6c337b78feddd3c1bfbd available at: https://gist.github.com/4f85bc103cdaf1cee2fed7452a2a40c4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10075-43e1cec.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10075-43e1cec.dmg

**Pull request merged**